src/PostgREST/App.hs

```diff
@@ -170,63 +174,73 @@ runDbHandler appState isoLvl mode authenticated prepared handler = do
 
   liftEither resp
 
-handleRequest :: AuthResult -> AppConfig -> AppState.AppState -> Bool -> Bool -> PgVersion -> ApiRequest -> SchemaCache -> Maybe ServerTimingParams -> Handler IO Wai.Response
-handleRequest AuthResult{..} conf appState authenticated prepared pgVer apiReq@ApiRequest{..} sCache serverTimingParams =
+handleRequest :: AuthResult -> AppConfig -> AppState.AppState -> Bool -> Bool -> PgVersion -> ApiRequest -> SchemaCache -> (ServerMetric, Double) -> Handler IO Wai.Response
+handleRequest AuthResult{..} conf appState authenticated prepared pgVer apiReq@ApiRequest{..} sCache jwtTime =
   case (iAction, iTarget) of
     (ActionRead headersOnly, TargetIdent identifier) -> do
-      wrPlan <- liftEither $ Plan.wrappedReadPlan identifier conf sCache apiReq
-      resultSet <- runQuery roleIsoLvl (Plan.wrTxMode wrPlan) $ Query.readQuery wrPlan conf apiReq
-      pgrst <- liftEither $ Response.readResponse wrPlan headersOnly identifier apiReq resultSet serverTimingParams
-      return $ pgrstResponse pgrst
+      (planTime, wrPlan) <- timeItT $ liftEither $ Plan.wrappedReadPlan identifier conf sCache apiReq
+      (rsTime, resultSet) <- timeItT $ runQuery roleIsoLvl (Plan.wrTxMode wrPlan) $ Query.readQuery wrPlan conf apiReq
+      (renderTime, pgrst) <- timeItT $ liftEither $ Response.readResponse wrPlan headersOnly identifier apiReq resultSet
```

Comment:
`timeItT` always gets done despite `configDbPlanEnabled=false` right? Ideally we don't time the action if it's not necessary.

Reply:
That what I was going to do, but this way it seemed to be easier to implement at one hand, and not introduce significant overhead due to the measurement, on the other. Do you have any particular improvements in mind?

Reply:
> and not introduce significant overhead due to the measurement

I'm not sure if this is the case. A more realistic load testing would settle it. 

We have this repo now: https://github.com/PostgREST/postgrest-benchmark. Try running the [quickstart](https://github.com/PostgREST/postgrest-benchmark/#quickstart) which includes a load test. Then:

- [produce a static bin](https://github.com/PostgREST/postgrest/tree/main/nix#building-postgrest) of this branch.
- use the static bin on the [postgrest derivation](https://github.com/PostgREST/postgrest-benchmark/blob/master/postgrest/postgrest.nix).
- rerun the load test and compare the results to the previous.

If they don't vary much, then I guess we can keep it as is.

---

Might be easier to do the `timeIt` conditionally than running the above benchmark :smile:

Reply:
Okay, made timing conditional, please review again